server/core/server-core.go

```diff
@@ -81,63 +77,105 @@ func validateMessage(message *messageService.Message, messageType int, conn *web
 	}
 
 	if message.Header.Type_ == "register" {
-
 		if _, ok := clients[message.Header.UserName]; ok {
 			loger.Log.Warn("User already exist")
 			return
 		} else {
-
-			clients[message.Header.UserName] = Client{conn: conn}
-
-		}
-		var x *messageService.User
-		json.Unmarshal(message.Body, &x)
-		return
-	}
-
-	if message.Header.Type_ == "auth" {
-		if _, ok := clients[message.Header.UserName]; ok {
-			loger.Log.Warn("User already exist")
-			return
-		} else {
-			// треба подумати над порівнянням токенів
-			//request_token := message.Header.Token
-			//ok := Client{token: request_token}
-			//if ok.token != request_token {
-			//	loger.Log.Errorf("Not valid token")
-			//	return
-			//}
-			var x *messageService.User
-			json.Unmarshal(message.Body, &x)
-			us, tok, err := auth.Login(x.UserName, x.Password)
+			var user *messageService.User
+			err := json.Unmarshal(message.Body, &user)
+			if err != nil {
+				loger.Log.Warn("failed to unmarshal body")
+				return
+			}
+			us, tok, err := auth.RegisterNewUser(user)
+			if err != nil {
+				loger.Log.Errorf("failed to register user", err)
+				return
+			}
 			newMessageHeader := messageService.MessageHeader{
 				Type_:    "authorization",
-				Command:  "loginissucc", //commands will be added to config file in near future
+				Command:  "registrissucc",
 				UserName: us.UserName,
 				Token:    tok,
 			}
-			// How to fill body message
-			newMessageBody := messageService.MessageBody{}
-
 			newMessage := messageService.Message{
 				Header: newMessageHeader,
-				Body:   newMessageBody,
 			}
-
-			text, err := messageService.MarshalMessage(&newMessage)
-			if err != nil {
-				loger.Log.Errorf("Marshal message failed")
-			}
-			err = conn.WriteMessage(websocket.TextMessage, text)
+			marshaledMessage, err := messageService.MarshalMessage(&newMessage)
 			if err != nil {
-				loger.Log.Errorf("Login failed")
+				loger.Log.Errorf("Can`t marshal message. %s", err)
+				return
+			} else {
+				if err := conn.WriteMessage(websocket.TextMessage, marshaledMessage); err != nil {
+					loger.Log.Errorf("Can not send message. %s", err)
+					return
+				}
 			}
-			clients[message.Header.UserName] = Client{conn: conn, token: tok}
 		}
+		clients[message.Header.UserName] = Client{conn: conn}
 		return
 	}
 
-	// TODO: Add token check
+	if message.Header.Type_ == "auth" {
+		if clientToken, ok := clients[message.Header.UserName]; ok {
+			loger.Log.Warn("User already exist")
+			return
+		} else {
+			if clientToken.token != message.Header.Token {
```

Comment:
Move all this check to separate method. It's already mess here.
As I remember, we talked about method `Verify` in `auth` package